sql: make RemoveIndexZoneConfigs a noop when run on behalf of a tenant

asubiotto opened the pull request:
RemoveIndexZoneConfigs would previously return an error when run on behalf of
a tenant because it attempts to write a zone config. This would cause some
operations like index truncation to fail unnecessarily.

Release note: None (multitenancy)

Fixes #52558

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52984)
<!-- Reviewable:end -->

rohany commented:
While you're here, you should also update `maybeUpdateZoneConfigsForPKChange`

asubiotto commented:
Done.

Commit:
sql: make RemoveIndexZoneConfigs a noop when run on behalf of a tenant

RemoveIndexZoneConfigs would previously return an error when run on behalf of
a tenant because it attempts to write a zone config. This would cause some
operations like index truncation to fail unnecessarily.

Release note: None (multitenancy)

asubiotto commented:
TFTRs

bors r=rohany,nvanbenschoten

asubiotto commented:
bors r=rohany,nvanbenschoten
